src/Solana/useSolana.tsx

```diff
@@ -0,0 +1,391 @@
+import { useCallback, useEffect, useState } from 'react'
+import { Buffer } from 'buffer'
+import {
+  Connection,
+  PublicKey,
+  clusterApiUrl,
+  Cluster,
+  VersionedTransaction,
+  AccountInfo,
+} from '@solana/web3.js'
+import { SolanaStatus, useSolanaStatus, useSolanaVars } from './solanaSentinel'
+import * as Currency from '@helium/currency-utils'
+import {
+  Asset,
+  heliumAddressToSolPublicKey,
+  searchAssets,
+  SearchAssetsOpts,
+  sendAndConfirmWithRetry,
+} from '@helium/spl-utils'
+import * as Hotspot from '@helium/hotspot-utils'
+import {
+  AccountLayout,
+  getAssociatedTokenAddress,
+  TOKEN_PROGRAM_ID,
+} from '@solana/spl-token'
+import {
+  entityCreatorKey,
+  init,
+  makerKey,
+} from '@helium/helium-entity-manager-sdk'
+import BigNumber from 'bignumber.js'
+import { getBalance } from '@helium/currency-utils'
+import axios from 'axios'
+import { AnchorProvider, Wallet, Program } from '@coral-xyz/anchor'
+import { daoKey } from '@helium/helium-sub-daos-sdk'
+import { HeliumEntityManager } from '@helium/idls/lib/types/helium_entity_manager'
+
+type Account = AccountInfo<string[]>
+
+// TODO: Get urls for each cluster
+const METAPLEX_URL = 'https://rpc-devnet.aws.metaplex.com/'
+
+export const createConnection = (cluster: Cluster) =>
+  new Connection(clusterApiUrl(cluster))
+
+const useSolana = ({
+  cluster: propsCluster = 'devnet',
+  heliumWallet,
+  solanaStatusOverride,
+}: {
+  cluster?: Cluster
+  heliumWallet?: string
+  solanaStatusOverride?: SolanaStatus
+}) => {
+  const { isHelium, isSolana, inProgress } =
+    useSolanaStatus(solanaStatusOverride)
+
+  const { data: vars } = useSolanaVars(propsCluster)
+
+  const [cluster, setCluster] = useState(propsCluster)
+  const [wallet, setWallet] = useState<PublicKey>()
+  const [connection, setConnection] = useState(createConnection(propsCluster))
+  const [hemProgram, setHemProgram] = useState<Program<HeliumEntityManager>>()
+
+  useEffect(() => {
+    if (!heliumWallet) return
+
+    // TODO: Is this right?
+    const anchorWallet = {
```

Comment:
The Wallet class should support signing.

Something like this:
```
    const anchorWallet = {
      signTransaction: async (transaction: Transaction) => {
        transaction.partialSign(secureAcct)
        return transaction
      },
      signAllTransactions: async (transactions: Transaction[]) => {
        return transactions.map((tx) => {
          tx.partialSign(secureAcct)
          return tx
        })
      },
      get publicKey() {
        return secureAcct?.publicKey
      },
    } as Wallet
    ```

Reply:
I guess since they don't have the private key then what you have is fine. I would imagine the maker would have to deep link to sign the transaction

Reply:
This should be fine so long as they never sign transactions here. Otherwise yeah you'd need to implement the deep link